src/cata_variant.h

```diff
@@ -21,9 +21,8 @@
 
 class JsonOut;
 class JsonValue;
-template <typename E> struct enum_traits;
-
 enum class mutagen_technique : int;
+template <typename E> struct enum_traits;
```

Comment:
IDK what triggered this but it's just a reordering.

Reply:
Likely a result of multiple passes of IWYU on the codebase, first one removing forward-declare, and adding `enum_traits.h` include, the second one undoing the change.
This is a thing that is kinda expected to happen with sweeping changes this large, when the codebase is very much in flux and has not reached steady state yet.

This should not be happening on "regular" prs, only maybe with other major refactors (and even then, kinda unlikely)